miniwin/miniwin/src/miniwin_d3drmframe.cpp

```diff
@@ -101,11 +143,18 @@ HRESULT Direct3DRMFrameImpl::GetVisuals(IDirect3DRMVisualArray** visuals)
 
 HRESULT Direct3DRMFrameImpl::SetTexture(IDirect3DRMTexture* texture)
 {
+	if (!texture) {
+		return DDERR_GENERIC;
+	}
+	auto textureImpl = static_cast<Direct3DRMTextureImpl*>(texture);
+	if (!textureImpl) {
+		return DDERR_GENERIC;
+	}
```

Comment:
Not having this check means we get nice ASAN errors when it's used incorrectly, This is helful since the return code is often not checked by the engine.

Reply:
This test will never fail, since we're using `static_cast` instead of `dynamic_cast`. (I originally used `dynamic_cast`)
So I think it can be removed.